### PR TITLE
[azure] marketplace image tests update

### PR DESCRIPTION
**Pull request by jinyunma**
As marketplace image issues have been solved by DPP https://issues.redhat.com/browse/DPP-13159,

1. enabled marketplace image job
2. change cluster profile from `azure-marketplace-qe` to `azure-qe`
3. rename job name to keep consistence with aws/gcp
4. rename step `ipi-conf-azure-workers-marketimage` to `ipi-conf-azure-osimage` to support marketplace image for all nodes (https://issues.redhat.com/browse/CORS-2725)

Notes:
as compatibility issue using 4.8 marketplace image to install 4.14 OCP cluster, change to use ARO image for master nodes instead. Once 4.13 image is available in azure marketplace, will adjust to use that image on masters.

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

**Comment by jinyunma** [comment]
> /pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

[job](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/42330/rehearse-42330-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure/1691307821972328448) passed.

install-config file
```
platform:
  azure:
    region: northcentralus
    baseDomainResourceGroupName: os4-common
controlPlane:
  architecture: amd64
  name: master
  platform:
    azure:
      type: null
compute:
- architecture: amd64
  name: worker
  replicas: 3
  platform:
    azure:
      type: Standard_D4s_v3
      osImage:
        publisher: redhat
        offer: rh-ocp-worker
        sku: rh-ocp-worker-gen1
        version: 4.8.2021122100
```

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei @MayXuQQ to review

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/marketplace-image/cucushift-installer-rehearse-azure-ipi-marketplace-image-workflow.metadata.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-	"path": "cucushift/installer/rehearse/azure/ipi/workers-marketimage/cucushift-installer-rehearse-azure-ipi-workers-marketimage-workflow.yaml",
+	"path": "cucushift/installer/rehearse/azure/ipi/marketplace-image/cucushift-installer-rehearse-azure-ipi-marketplace-image-workflow.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/marketplace-image/cucushift-installer-rehearse-azure-ipi-marketplace-image-workflow.metadata.json:2]: how about renaming `marketplace-image` to `marketplace` to keep consistent with gcp and aws platform.

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/marketplace-image/cucushift-installer-rehearse-azure-ipi-marketplace-image-workflow.metadata.json]: sure, updated.

File: ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  # image plan is case-sensitive, make sure that publisher/offer/sku keep the same as plan 
+  if [[ "${OS_IMAGE_MASTERS_PLAN}" == "WithPurchasePlan" ]] || [[ "${OS_IMAGE_MASTERS_PLAN}" == "" ]]; then
+    imageInfo[0]=$(az vm image show --urn ${image} --query 'plan.publisher' -otsv)
+    imageInfo[1]=$(az vm image show --urn ${image} --query 'plan.product' -otsv)
+    imageInfo[2]=$(az vm image show --urn ${image} --query 'plan.name' -otsv)
+  fi
+
+  # create a patch to set osImage for control plane instances
+  PATCH_MASTER="/tmp/install-config-master-marketimage.yaml.patch"
+  cat > "${PATCH_MASTER}" << EOF
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh:96]: @jinyunma per yesterday's debugging, so far, we only can see 4.8 boot image, so the 4.14 install would failed as expected, right?

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**